src/libraries/System.Runtime/tests/System/Runtime/ControlledExecutionTests.cs

```diff
@@ -0,0 +1,199 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading;
+using Xunit;
+
+// Disable warnings for ControlledExecution.Run
+#pragma warning disable SYSLIB0046
+
+namespace System.Runtime.Tests
+{
+    public class ControlledExecutionTests
+    {
+        private bool _startedExecution, _caughtException, _finishedExecution;
+        private Exception _exception;
+        private CancellationTokenSource _cts;
+        private volatile int _counter;
+
+        // Tests cancellation on timeout. The ThreadAbortException must be mapped to OperationCanceledException.
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsWindows), nameof(PlatformDetection.IsNotNetFramework), nameof(PlatformDetection.IsNotMonoRuntime), nameof(PlatformDetection.IsNotNativeAot))]
```

Comment:
`PlatformDetection.IsNotNetFramework` is unnecessary here. System.Runtime tests do not run on .NET Framework.